File: system-test/typescript/src/testkit/dsl/port/when/steps/when-place-order.ts
import type { WhenStep } from './when-step.js';

export interface WhenPlaceOrder extends WhenStep {
  withOrderNumber(orderNumber: string): WhenPlaceOrder;
  withSku(sku: string): WhenPlaceOrder;
  withQuantity(quantity: string | number): WhenPlaceOrder;
  withCountry(country: string): WhenPlaceOrder;
  withCouponCode(couponCode?: string): WhenPlaceOrder;
}

File: system/multitier/backend-clean-java/src/testSupport/java/com/mycompany/myshop/backend/testkit/dsl/port/when/steps/WhenPlaceOrder.java
package com.mycompany.myshop.backend.testkit.dsl.port.when.steps;

import com.mycompany.myshop.backend.testkit.dsl.port.when.steps.base.WhenStep;

public interface WhenPlaceOrder extends WhenStep {
    WhenPlaceOrder withSku(String sku);

    WhenPlaceOrder withQuantity(int quantity);

    WhenPlaceOrder withQuantity(String quantity);

    WhenPlaceOrder withCountry(String country);

    WhenPlaceOrder withCouponCode(String couponCode);

    WhenPlaceOrder withCouponCode();
}
